util/src/sync/mutex.rs

```diff
@@ -0,0 +1,126 @@
+use super::atomic::{spin_loop_hint, AtomicBool, Ordering};
+use crate::cell::CausalCell;
+use core::{fmt, ops};
+
+/// A simple spinlock ensuring mutual exclusion.
+#[derive(Debug)]
+pub struct Mutex<T> {
```

Comment:
I wonder whether it'd be worthwhile to call this `Spinlock<T>` instead of `Mutex<T>`, to make it clear that it has a spinlock's performance characteristics. It's probably already pretty clear, given that we're in a kernel & don't have a scheduler.

Reply:
That’s not a bad call! I was thinking about writing a read-write lock too, though

Reply:
We could have `SpinLock<T>` (`SpinMutex<T>`?) and `SpinRwlock<T>` if needed. I don't have super strong feelings about this, though.

Reply:
They could also go in a `spin` module, though I don't imagine we'd ever have any _other_ kind of mutex. I don't feel passionately about this either.